plugin/helpers_kv.go

```diff
@@ -129,3 +130,91 @@ func (p *HelpersImpl) KVSetWithExpiryJSON(key string, value interface{}, expireI
 
 	return nil
 }
+
+type kvListOptions struct {
+	KVListPerPage int
+	Prefix        string
+	Checker       func(key string) (keep bool, err error)
+}
+
+// KVListOption represents a single input option for KVListWithOptions
+type KVListOption func(*kvListOptions)
+
+// WithPrefix will filter out any key that doesn't start with the given string.
+func WithPrefix(prefix string) KVListOption {
+	return func(args *kvListOptions) {
+		args.Prefix = prefix
+	}
+}
+
+// WithChecker allows for a custom filter function to determine which keys to return.
+// Returning true will keep the key and false will filter it out.  Returning an error
+// will halt KVListWithOptions immediately and pass the error up (with no other results).
+func WithChecker(f func(key string) (keep bool, err error)) KVListOption {
+	return func(args *kvListOptions) {
+		args.Checker = f
+	}
+}
+
+// WithKVListPerPage is for unit testing KVListWithOptions.  This value defaults to 100 and shouldn't
+// be overriden outside of testing.
+func WithKVListPerPage(count int) KVListOption {
+	return func(args *kvListOptions) {
+		args.KVListPerPage = count
+	}
+}
+
+// KVListWithOptions implements Helpers.KVListWithOptions.
+func (p *HelpersImpl) KVListWithOptions(options ...KVListOption) ([]string, error) {
+	err := p.ensureServerVersion("5.6.0")
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
nit 0/5: remove these few blank lines - initialization is a single "paragraph"

Reply:
I think I've now set the paragraph break where you want it.